File: app/src/main/java/com/shen/xi/android/tut/ImageViewFragment.java
package com.shen.xi.android.tut;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.google.inject.Inject;
import com.nostra13.universalimageloader.core.ImageLoader;
import com.nostra13.universalimageloader.core.assist.SimpleImageLoadingListener;
import com.squareup.otto.Bus;

import uk.co.senab.photoview.PhotoViewAttacher;


public class ImageViewFragment extends Fragment {
    private static final String ARG_ORDER = "weibo status position in context";
    @Inject
    private ImageLoader mImageLoader;
    private PhotoViewAttacher mPhotoViewAttacher;
    private int mOrder;
    @Inject
    private Bus mBus;

    public ImageViewFragment() {
        TuTModule.getInjector().injectMembers(this);
    }

    /**
     * create a new instance of {@link ImageViewFragment}
     *
     * @param order the order of the item in its parent context
     * @return a new instance of {@link ImageViewFragment}
     */
    public static Fragment newInstance(int order) {
        ImageViewFragment fragment = new ImageViewFragment();
        Bundle bundle = new Bundle();
        bundle.putInt(ARG_ORDER, order);

        fragment.setArguments(bundle);

        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle arguments = getArguments();
        if (arguments != null) {
            mOrder = arguments.getInt(ARG_ORDER);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.image_view_full, container, false);

        ImageView imageView = (ImageView) view.findViewById(android.R.id.content);

        mImageLoader.displayImage(((AbstractImageViewActivity) getActivity()).getImageUrlByOrder(
                                          mOrder), imageView,
                                  new SimpleImageLoadingListener() {
                                      @Override
                                      public void onLoadingComplete(String imageUri, View view,
                                                                    Bitmap loadedImage) {
                                          ImageView imageView = (ImageView) view;
                                          imageView.setImageBitmap(loadedImage);

                                          mPhotoViewAttacher = new PhotoViewAttacher(imageView);
                                      }
                                  }
        );

        return view;
    }

    @Override
    public void onStop() {
        if (mPhotoViewAttacher != null)
            mPhotoViewAttacher.cleanup();

        super.onStop();
    }

}

File: app/src/main/scala/com/shen/xi/android/tut/ImageViewFragment.scala
package com.shen.xi.android.tut

import android.graphics.Bitmap
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.{LayoutInflater, View, ViewGroup}
import android.widget.ImageView
import com.google.inject.Inject
import com.nostra13.universalimageloader.core.ImageLoader
import com.shen.xi.android.tut.util.MySimpleImageLoadingListener
import com.squareup.otto.Bus
import uk.co.senab.photoview.PhotoViewAttacher

object ImageViewFragment {
  private val ARG_ORDER = "weibo status position in context"

  /**
   * create a new instance of 'ImageViewFragment'
   *
   * @param order the order of the item in its parent context
   * @return a new instance of { @link ImageViewFragment}
   */
  def newInstance(order: Int) = {
    val fragment = new ImageViewFragment()
    val bundle = new Bundle()
    bundle.putInt(ARG_ORDER, order)

    fragment.setArguments(bundle)

    fragment
  }
}

class ImageViewFragment extends Fragment {

  import com.shen.xi.android.tut.ImageViewFragment.ARG_ORDER

  @Inject
  private var mImageLoader: ImageLoader = null
  private var mPhotoViewAttacher: PhotoViewAttacher = null
  private var mOrder: Int = -1
  @Inject
  private var mBus: Bus = null


  TuTModule.getInjector.injectMembers(this)


  override def onCreate(savedInstanceState: Bundle) = {
    super.onCreate(savedInstanceState)

    val arguments = getArguments
    if (arguments != null) {
      mOrder = arguments.getInt(ARG_ORDER)
    }
  }

  override def onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle) = {
    // Inflate the layout for this fragment
    val view = inflater.inflate(R.layout.image_view_full, container, false)
    val imageView = view.findViewById(android.R.id.content).asInstanceOf[ImageView]

    mImageLoader.displayImage(
      getActivity.asInstanceOf[AbstractImageViewActivity].getImageUrlByOrder(mOrder),
      imageView,
      new MySimpleImageLoadingListener(
        container.getMeasuredWidth,
        container.getMeasuredHeight,
        (_, view: View, _) => mPhotoViewAttacher = new PhotoViewAttacher(view.asInstanceOf[ImageView])
      )
    )

    view
  }

  override def onStop() = {
    if (mPhotoViewAttacher != null)
      mPhotoViewAttacher.cleanup()

    super.onStop()
  }

}
